resolwe/flow/managers/base.py

```diff
@@ -211,7 +218,18 @@ def __init__(self, *args, **kwargs):
         self._drop_ctypes = getattr(settings, 'FLOW_MANAGER_DISABLE_CTYPE_CACHE', False)
 
         if 'static' not in kwargs or not kwargs['static']:
-            super().__init__(*args, **kwargs)
+            # Update global singleton to point to the current instance. Otherwise,
+            # the global instance will be completely incorrect in Channels workers.
+            from resolwe.flow.managers import override_manager
+
+            with override_manager(self):
```

Comment:
@jberci can this be called in parallel in case of multi-threaded Channels workers? In this case, the global manager singleton is not thread-safe and this new override I introduced is problematic.

Reply:
Oh, it seems that Channels workers are single-threaded.